pubsub/gcppubsub/gcppubsub.go

```diff
@@ -210,8 +211,63 @@ func (o *URLOpener) OpenTopicURL(ctx context.Context, u *url.URL) (*pubsub.Topic
 
 // OpenSubscriptionURL opens a pubsub.Subscription based on u.
 func (o *URLOpener) OpenSubscriptionURL(ctx context.Context, u *url.URL) (*pubsub.Subscription, error) {
-	for param := range u.Query() {
-		return nil, fmt.Errorf("open subscription %v: invalid query parameter %q", u, param)
+	// Set subscription options to use defaults
+	o.SubscriptionOptions = SubscriptionOptions{
+		ReceiveMaxHandlers:  recvBatcherOpts.MaxHandlers,
```

Comment:
If we do this, let's rename `recvBatchOpts` -> `defaultRecvBatchOpts` to make clear it's not used explicitly.

In fact, if you do that you fill find that it *is* used explicitly, on line 519; that will need to be updated to use the saved `SubscriptionOptions`.